File: analytics/src/main/java/io/appmetrica/analytics/impl/utils/validation/ValidationResultsChain.java
package io.appmetrica.analytics.impl.utils.validation;

import androidx.annotation.Nullable;
import java.util.LinkedList;
import java.util.List;
import kotlin.collections.CollectionsKt;

public class ValidationResultsChain implements Validator<List<ValidationResult>> {

    @Override
    public ValidationResult validate(@Nullable List<ValidationResult> results) {
        LinkedList<String> errorMessages = new LinkedList<String>();
        boolean valid = true;
        for (ValidationResult result : results) {
            if (result.isValid() == false) {
                errorMessages.add(result.getDescription());
                valid = false;
            }
        }
        return valid ? ValidationResult.successful(this)
                : ValidationResult.failed(this, CollectionsKt.joinToString(errorMessages, ", ", "", "", -1, "", null));
    }
}

File: core-utils/src/main/java/io/appmetrica/analytics/coreutils/internal/validation/ValidationResultsChain.kt
package io.appmetrica.analytics.coreutils.internal.validation

class ValidationResultsChain : Validator<@JvmSuppressWildcards List<ValidationResult>> {

    override fun validate(data: List<ValidationResult>): ValidationResult {
        val errors = data.filter { !it.isValid }.map { it.description }
        return if (errors.isEmpty()) {
            ValidationResult.successful(this)
        } else {
            ValidationResult.failed(this, errors.joinToString(", "))
        }
    }
}
